package.json

```diff
@@ -183,7 +184,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "yarn precommit"
+      "pre-commit": "yarn typecheck && yarn precommit"
```

Comment:
Doing it like this because `tsc` does not take files as input, so it cannot be used by `lint-staged`